src/Exception/ClientInitialisationException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+namespace Microsoft\Graph\Exception;
+
+/**
+ * Class ClientInitialisationException
+ *
+ * @package Microsoft\Graph\Exception
+ * @copyright 2021 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+class ClientInitialisationException extends GraphException
+{
```

Comment:
Why is this empty?

Reply:
It extends all functionality of the GraphException with nothing custom to add.

Thanks for pointing out a bug though, I need to change `GraphException`'s `__toString()` to print the child class name where needed

Reply:
What is the purpose of GraphException class? 

For the JS SDK, @MIchaelMainer had suggested creating a GraphCientError class which handles the errors such as  ClientInitialisationException and a GraphError class which handles errors from the API.

I would suggest using something similar because:
- There is a sync in the SDKs
- Avoid creating an empty class if you are not extending or adding more functionalities. 
- A generic GraphClientError class with clear messages reduces the need to create multiple class depending on the feature. For example, ClientInitialization,  an error class for PageIteration.

Reply:
Thank you for catching this! We still need to better define our error classification. At a high-level, we have two primary exceptions: ServiceException that represents errors returned in an API response and ClientException for issues raised in the client. Both of these can extend GraphException.

I just created a quick PR to help us be more consistent. 

https://github.com/microsoftgraph/msgraph-sdk-design/pull/47

Reply:
Updated ClientInitialisationException to GraphClientException

Reply:
@Ndiritu Graph Exception that is the parent class is it the service  exception class?
What is the class which captures the Graph API error response? 

If GraphException is the service error response class then I  recommend you to  keep them separate as mentioned by @MIchaelMainer in the PR https://github.com/microsoftgraph/msgraph-sdk-design/pull/47

I suppose you can address this in a separate PR focusing on the error classes.

Reply:
@nikithauc Yes, i'll need to make these changes in a separate PR.